File: D/python/code.py
from typing import List


def get_longest_increasing_path(matrix: List[List[int]]) -> int:
    # your code goes here
    return 0

def read_matrix() -> List[List[int]]:
    n, m = map(int, input().split())
    matrix = []
    for i in range(n):
        matrix.append(list(map(int, input().split())))
    return matrix

matrix = read_matrix()
print(get_longest_increasing_path(matrix))

File: D/js/code.js
function getLongestIncreasingPath(matrix) {
    // your code goes here
    return 0;
}


const _readline = require('readline');

const _reader = _readline.createInterface({
    input: process.stdin
});

const _inputLines = [];
let _curLine = 0;

_reader.on('line', line => {
    _inputLines.push(line);
});

process.stdin.on('end', solve);


function solve() {
    const matrix = readMatrix();
    const ans = getLongestIncreasingPath(matrix);
    console.log(ans);    
}

function readInt() {
    const n = Number(_inputLines[_curLine]);
    _curLine++;
    return n;
}

function readArray() {
    var arr = _inputLines[_curLine].trim(" ").split(" ").map(num => Number(num));
    _curLine++;
    return arr;
}

function readMatrix() {
    let sizes = readArray();
    let n = sizes[0];
    let matrix = [];
    for (let i = 0; i < n; i++) {
        matrix.push(readArray());
    }
    return matrix;
}